apps/transport/lib/transport/dataset.ex

```diff
@@ -207,21 +208,37 @@ defmodule Transport.Dataset do
   def filter_has_realtime, do: from d in __MODULE__, where: d.has_realtime == true
   def count_has_realtime, do: Repo.aggregate(filter_has_realtime(), :count, :id)
 
-  @spec get_by([{:slug, any}, ...]) :: Transport.Dataset.t | nil
-  def get_by(slug: slug) do
-    __MODULE__
-    |> where(slug: ^slug)
-    |> preload_without_validations()
+  @spec get_by(keyword) :: Dataset.t()
+  def get_by(options) do
```

Comment:
why don't you use argument destructuring there ?

Reply:
It's a Keyword; not a map.
With keywork you can map just the first item of the list.